src/Type/User/Mutation/UserMutation.php

```diff
@@ -0,0 +1,209 @@
+				],
+				'nickname'    => [
+					'type'        => Types::string(),
+					'description' => __( 'The user\'s nickname, defaults to the user\'s username.', 'wp-graphql' ),
+				],
+				'firstName'   => [
+					'type'        => Types::string(),
+					'description' => __( '	The user\'s first name.', 'wp-graphql' ),
+				],
+				'lastName'    => [
+					'type'        => Types::string(),
+					'description' => __( 'The user\'s last name.', 'wp-graphql' ),
+				],
+				'description' => [
+					'type'        => Types::string(),
+					'description' => __( 'A string containing content about the user.', 'wp-graphql' ),
+				],
+				'richEditing' => [
+					'type'        => Types::string(),
+					'description' => __( 'A string for whether to enable the rich editor or not. False if not empty.', 'wp-graphql' ),
+				],
+				'registered'  => [
+					'type'        => Types::string(),
+					'description' => __( 'The date the user registered. Format is Y-m-d H:i:s.', 'wp-graphql' ),
+				],
+				'role'        => [
+					'type'        => Types::string(),
+					'description' => __( 'A string used to set the user\'s role.', 'wp-graphql' ),
+				],
+				'jabber'      => [
+					'type'        => Types::string(),
+					'description' => __( 'User\'s Jabber account.', 'wp-graphql' ),
+				],
+				'aim'         => [
+					'type'        => Types::string(),
+					'description' => __( 'User\'s AOL IM account.', 'wp-graphql' ),
+				],
+				'yim'         => [
+					'type'        => Types::string(),
+					'description' => __( 'User\'s Yahoo IM account.', 'wp-graphql' ),
+				],
+				'locale'      => [
+					'type'        => Types::string(),
+					'description' => __( 'User\'s locale.', 'wp-graphql' ),
+				],
+			];
+
+			/**
+			 * Filters all of the fields available for input
+			 *
+			 * @var array $input_fields
+			 */
+			self::$input_fields = apply_filters( 'graphql_user_mutation_input_fields', $input_fields );
+
+		}
+
+		return ( ! empty( self::$input_fields ) ) ? self::$input_fields : null;
+
+	}
+
+	/**
+	 * Maps the GraphQL input to a format that the WordPress functions can use
+	 *
+	 * @param array $input Data coming from the GraphQL mutation query input
+	 * @param string $mutation_name Name of the mutation being performed
+	 * @access public
+	 * @return array
+	 */
+	public static function prepare_user_object( $input, $mutation_name ) {
+
+		$insert_user_args = [];
+
+		if ( ! empty( $input['password'] ) ) {
+			$insert_user_args['user_pass'] = $input['password'];
+		}
+
+		if ( ! empty( $input['login'] ) ) {
+			$insert_user_args['user_login'] = $input['login'];
+		}
+
+		if ( ! empty( $input['nicename'] ) ) {
+			$insert_user_args['user_nicename'] = $input['nicename'];
+		}
+
+		if ( ! empty( $input['websiteUrl'] ) ) {
+			$insert_user_args['user_url'] = esc_url( $input['websiteUrl'] );
+		}
+
+		if ( ! empty( $input['email'] ) ) {
```

Comment:
turns out that if you enter a plain string instead of an email address for email, the email just gets set to null. . .

So, I'm thinking we need some validation here to ensure the input is an acceptable email address (probably need to dig and see what core is doing to validate the email when the update is made) before we allow it to be used as the `user_email` field for insertion.

I'm thinking, if the email isn't "valid" we just unset it and don't pass it to the mutation at all. . .and/or we throw an exception so the user knows the email was invalid.

See attached GIF. 

![string-emails-not-working](https://user-images.githubusercontent.com/1260765/29326634-92f49d4e-81a9-11e7-8383-9b60e6353c1e.gif)

Reply:
I think the exception route would be best for now. Until someone has an objection to doing this, I say we roll with that.

Reply:
Ya, let's do some validation on the input to make sure it's an email and throw an exception if it's not proper